pkg/controller/service/reconciler.go

```diff
@@ -135,8 +135,7 @@ func (r *Reconciler) remediateAllocatedNodePorts(ctx context.Context, log logr.L
 				port.NodePort == nodePortIngress ||
 				port.NodePort == nodePortIstioIngressGatewayZone0 ||
 				port.NodePort == nodePortIstioIngressGatewayZone1 ||
-				port.NodePort == nodePortIstioIngressGatewayZone2 ||
-				port.NodePort == nodePortVirtualGardenKubeAPIServer {
```

Comment:
What about the other usages of `keyVirtualGardenKubeAPIServer` or `nodePortVirtualGardenKubeAPIServer`? Can they be cleaned up as well?

Reply:
You're right. We can directly switch to waiting for the `istio-ingressgateway` service and disregard the `virtual-garden-kube-apiserver` service 👍